src/rgw/rgw_data_sync.cc

```diff
@@ -1327,6 +1372,11 @@ class RGWDataSyncShardCR : public RGWCoroutine {
                                      1 /* no buffer */);
       error_repo->get();
       spawn(error_repo, false);
+      active_repo = new RGWOmapAppend(sync_env->async_rados, sync_env->store, 
+                                      rgw_raw_obj(pool, status_oid),
+                                      1);
+      active_repo->get();
+      spawn(active_repo, false);
```

Comment:
@cbodley use active_repo to record the active bucket shard shall not be necessary, but maybe it's also expensive and a little complex to read all the remote datalog entry so that to get the whole lagging bucket shards.